### PR TITLE
STANFORD-417 | Interactive Photo Card

### PR DESCRIPTION
**Pull request by yvonnetangsu**
# Ready for review

### CODE REVIEW

**Comment by yvonnetangsu** [comment]
@broleomo @iamrentman @JBCSU This is no where near ready (just setting up). Just want to test out opening and PR and see if this is how you'd like us to do things?

**Comment by yvonnetangsu** [comment]
@broleomo and @iamrentman Tagging both of you for an initial review. We can talk more about procedures (eg, do I deploy this) and other things next week in our sprint meeting :)

**Comment by dlcousins** [comment]
This is looking great @yvonnetangsu 

In the client.jsx (the front end renderer),  we have started to check if the component DOM node exists, as they are throwing JS errors when they do not. 

Snippet of code that would essentially replace your client.jsx file:

`
import hydrateComponent from "../../packages/utils/hydrate-component";
import Component from "./Component";
(function () {
  const componentName = "your-component-name";
  const target = document.querySelector(
    `[data-hydration-component="${componentName}"]`
  );

  if (!target) return;
  hydrateComponent({ Component, componentName });
})();
`

**Comment by yvonnetangsu** [comment]
> This is looking great @yvonnetangsu
> 
> In the client.jsx (the front end renderer), we have started to check if the component DOM node exists, as they are throwing JS errors when they do not.
> 
> Snippet of code that would essentially replace your client.jsx file:
> 
> import hydrateComponent from "../../packages/utils/hydrate-component"; import Component from "./Component"; (function () { const componentName = "your-component-name"; const target = document.querySelector( `[data-hydration-component="${componentName}"]` );
> 
> if (!target) return; hydrateComponent({ Component, componentName }); })();
> 
> Edit: The JSX tags are breaking the code block!

Thank you for the tip @dlcousins! Will update tomorrow 😄

File: components/interactive-photo-card/Component.jsx [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+            />
+            <button
+              type="button"
+              aria-hidden={!isFlipped}
+              tabIndex={isFlipped ? undefined : -1}
+              onClick={() => flipCard(0)}
+              aria-label="Dismiss content"
+              className="su-block su-ml-auto su-mr-0 su-border-3 su-border-white su-rounded-full su-text-white su-p-7 lg:su-p-14 su-stretched-link"
+            >
+              <ArrowsRotate className="su-size-30 lg:su-size-36 su-fill-none group-aria-hidden/back:invisible" />
```
Comment (iamrentman) [inline components/interactive-photo-card/Component.jsx:97]: @yvonnetangsu We discussed on our call, but add a little rotation as hover effect here

File: components/interactive-photo-card/Component.jsx [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+            className="su-group/front su-relative su-bg-white su-backface-hidden su-rounded-[8px] su-shadow-lg su-min-w-full"
+            aria-hidden={isFlipped}
+          >
+            <div className="su-flex su-flex-col su-h-full su-rs-px-5 su-rs-pt-6 su-rs-pb-4">
+              {eyebrow && (
+                <div className="su-type-1 su-text-black-60 su-font-semibold su-rs-mb-1">
+                  {eyebrow}
+                </div>
+              )}
+              <h3 className="su-grow su-type-4 su-font-bold su-font-sans su-type su-rs-mb-0">
```
Comment (iamrentman) [inline components/interactive-photo-card/Component.jsx:64]: On our call we said `h2` here instead.

File: components/interactive-photo-card/Component.jsx (hunk 1 above)
Comment (yvonnetangsu) [reply components/interactive-photo-card/Component.jsx]: @iamrentman hocus states added along with the heading level change 👍🏼 😄  If you could pull down for a quick look, after that I'll deploy it to component service this afternoon.

**Comment by yvonnetangsu** [comment]
FYI @broleomo and @iamrentman - I have this card behave so that the height of the card will expand to accommodate the full height of the content of the longer side (front or back), so if the content on the back is much longer than the heading in the front, it will still take on the height of the longer side so that when the card is flipped, it doesn't cause a layout jump 😄 

This is the trick I'm using here:
https://www.smashingmagazine.com/2020/02/magic-flip-cards-common-sizing-problem/

**Comment by yvonnetangsu** [comment]
I've deployed the component and added it to the Stanford University set. Please let me know what else I need to do. Thanks! @dlcousins @hmalcolmSquiz cc @iamrentman @broleomo 
https://dxp.squiz.cloud/organization/sug-9278/component-service/component-sets/su-cs/

**Comment by yvonnetangsu** [comment]
@iamrentman @broleomo Per discussion in the pair programming session, I'm going to go ahead and merge this into `develop` 😄

**Pull request merged**